src/main/java/com/faforever/client/io/FeaturedModFileCacheService.java

```diff
@@ -0,0 +1,118 @@
+package com.faforever.client.io;
+
+import com.faforever.client.api.dto.FeaturedModFile;
+import com.faforever.client.preferences.PreferencesService;
+import com.faforever.client.task.ResourceLocks;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.nio.file.attribute.FileTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.util.stream.Stream;
+
+
+@Service
+@Slf4j
+public class FeaturedModFileCacheService implements InitializingBean {
+  private final Path cacheDirectory;
+  private final int cacheLifeTimeInDays;
+
+  public FeaturedModFileCacheService(PreferencesService preferencesService) {
+    this.cacheDirectory = preferencesService.getCacheDirectory();
+    this.cacheLifeTimeInDays = preferencesService.getPreferences().getCacheLifeTimeInDays();
+  }
+
+  public boolean isCached(FeaturedModFile featuredModFile) {
+    return Files.exists(getCachedFilePath(featuredModFile));
+  }
+
+  private String readHashFromFile(Path filePath) {
+    // see buildCachedFileName
+    return filePath.getFileName().toString().split("\\.")[3];
+  }
+
+  private String buildCachedFileName(FeaturedModFile featuredModFile) {
+    return String.format(
+        "%s.%s.%s.%s",
```

Comment:
. seems to be a dangrous delimiter. Could there not be mod files that have a . in the name 🤔

Reply:
is that possible?

Reply:
And yes file name as almost always a . in it.... But anyway what do you need the file name for???